lab-adam/test/city-route-test.js

```diff
@@ -0,0 +1,295 @@
+          if (err) return done (err);
+          expect(res.status).to.equal(200);
+          for (var i in exampleCity) {
+            if (i === 'id') continue;
+            expect(res.body.i).to.equal(exampleCity.i);
+          }
+          expect(res.body.hospitals.length).to.equal(1);
+          expect(res.body.hospitals[0].name).to.equal('FakeHospital');
+          this.tempCity = res.body;
+          done();
+        });
+      });
+    });
+    describe('with invalid ID', function() {
+      it('should return a 404', done => {
+        request.get(`${url}/api/city/1234512312`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+    describe('GET with no ID to /api/city/ and pagination', function(){
+      before(done => {
+        var i = 0;
+        function createObjects(){
+          exampleCity.timestamp = new Date();
+          i++;
+          if(i >= 300) return;
+          new City(exampleCity).save()
+          .then(createObjects(i));
+        }
+        createObjects();
+        done();
+      });
+      after(done => {
+        delete exampleCity.timestamp;
+        if(this.tempCity){
+          City.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should return 50 cities', done => {
+        request.get(`${url}/api/city`)
+        .end((err, res) => {
+          // expect(res.body).to.be.instanceOf(Array);
+          expect(res.body).to.be.length(50);
+          done();
+        });
+      });
+      it('should return 100 cities', done => {
+        request.get(`${url}/api/city?pagesize=100`)
+        .end((err, res) => {
+          // expect(res.body).to.be.instanceOf(Array);
+          expect(res.body).to.be.length(100);
+          done();
+        });
+      });
+    });
+  });
+  describe('testing PUT requests', function(){
+    describe('with valid body', function(){
+      before(done => {
+        exampleCity.timestamp = new Date();
+        new City(exampleCity).save()
+        .then(city => {
+          this.tempCity = city;
+          done();
+        })
+        .catch(done);
+      });
+      after(done => {
+        delete exampleCity.timestamp;
+        if(this.tempCity){
+          City.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should return an updated city object', done => {
+        request.put(`${url}/api/city/${this.tempCity.id}`)
+        .send(updateCity)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          for (var i in updateCity) {
+            if (i !== '_id') expect(res.body[i]).to.equal(updateCity[i]);
+          }
+          this.tempCity = res.body;
+          done();
+        });
+      });
+    });
+    describe('with invalid or no body', function() {
+      before(done => {
```

Comment:
Your test environment could be defined in the outer describe block for all PUT requests, valid and invalid, (provided that you use proper lexical scoping) so that you don't repeat yourself.